moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -448,26 +448,30 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     return *joint_state_target_;
   }
 
+  void setStartState(const moveit_msgs::RobotState& start_state)
+  {
+    considered_start_state_ = start_state;
+  }
+
   void setStartState(const moveit::core::RobotState& start_state)
   {
-    considered_start_state_ = std::make_shared<moveit::core::RobotState>(start_state);
+    considered_start_state_ = moveit_msgs::RobotState();
+    moveit::core::robotStateToRobotStateMsg(start_state, *considered_start_state_, true);
   }
 
   void setStartStateToCurrentState()
   {
-    considered_start_state_.reset();
+    considered_start_state_ = std::nullopt;
```

Comment:
Instead of using std::optional, you could directly store an empty diff in `considered_start_state_`. This will simplify code when setting the request's start_state.

Reply:
Done :+1: